### DISCUSSION TITLE
Can I modify the number of edges?

### DISCUSSION THREAD

**Discussion by bosxered** [idaholab/moose#24913] [Q&A Meshing] [answered]
Hi Moose experts,

I want to modify the number of edges.
Following is my input file.

```
[Mesh]
  [./poly1]
    type = PolyLineMeshGenerator
    points = '0.0 0.0 0.0
              9.0 0.0 0.0
              9.0 1.1 0.0'
    loop = true
    num_edges_between_points = 9
  []
  [./tri]
    type = XYDelaunayGenerator
    boundary = 'poly1'
#    refine_boundary = true
    desired_area = 0.6
    output_subdomain_name = 'tri'
  []
[]
```

I used `num_edges_between_points = 9` so 10 edges are generated on EVERY side.
**Can I modify (10, 10, 10) edges to (10, 3, 10) edges?**

In other words, I want to make it
from this
![image](https://github.com/idaholab/moose/assets/79071957/b5b5b8e2-5854-42a0-8b5b-0eb6dd264cbf)

to this
![image](https://github.com/idaholab/moose/assets/79071957/d4731f8c-c859-4332-8d4f-178f6b1a6381)

For your information, the reason why I want to modify is to stitch with another subdomain.
If I don't modify it, hanging nodes that cannot calculate physics in the normal FEM method are generated so that I want to avoid this situation.

Thank you.

Best Regards,
Jeonghwan

**Comment by GiudGiud** [comment]
Hello

Use two polyline MG, one for the 10 edges part and one for the single edge part, then stitch them
Then triangulate.


Guillaume

**Reply by bosxered** [reply]
Hello,

I am so sorry but could you please explain in more detail?
I tried your suggestion but it failed.

Things that I tried:
1. test/tests/meshgenerators/xy_delaunay_generator/xydelaunay_multiline.i
- I think the test file has problems, too

2.  I added one more line.
```
  [line3]
    type = PolyLineMeshGenerator
    points = '0.0 1.0 0.0
              0.0 0.0 0.0'
#    loop = true
  []

  [outer_bdy]
    type = StitchedMeshGenerator
    inputs = "line1 line2 line3"
    stitch_boundaries_pairs = '2 1;
                               2 1'
  []
```
Stitching was done successfully, but XYDelaunayG didn't work with this error.
```
*** ERROR ***
TriangulatorInterface does not support overlapping nodes found at (x,y,z)=(       0,        0,        0)
Stack frames: 15
0: libMesh::print_trace(std::ostream&)
1: libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*, std::ostream&)
2: libMesh::TriangulatorInterface::elems_to_segments()
3: libMesh::Poly2TriTriangulator::triangulate()
4: XYDelaunayGenerator::generate()
5: MeshGenerator::generateInternal()
6: MooseApp::executeMeshGenerators()
7: Action::timedAct()
8: ActionWarehouse::executeActionsWithAction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
9: ActionWarehouse::executeAllActions()
10: MooseApp::runInputFile()
11: MooseApp::run()
12: main
13: __libc_start_main
14: test9-opt() [0x402b30]
[0] ../src/mesh/triangulator_interface.C, line 106, compiled Feb 17 2023 at 11:44:58
```

Jeonghwan

**Reply by GiudGiud** [reply, marked as answer]
This doesnt work because you only stitch in two locations, but there are 3 connection points.
You need only 2 lines. 
One that does the angle, and one that does the right edge. 
Then you stitch them in a single StitchMeshGenerator at 2 locations (with two stitch pairs)

**Reply by bosxered** [reply]
Thank you so much!!
